common/utils/appinsights.js

```diff
@@ -0,0 +1,27 @@
+const appInsights = require('applicationinsights')
+const { applicationInsights } = require('../config')
+
+const initApplicationInsights = () => {
+  if (applicationInsights.disabled) {
+    console.log('Application Insights disabled; disable flag set')
+    return
+  }
+
+  if (applicationInsights.instrumentationKey === '') {
+    console.log('Applciation Insights disabled; no instrumentation key set')
+    return
+  }
+
+  appInsights
+    .setup(applicationInsights.instrumentationKey)
+    .setDistributedTracingMode(appInsights.DistributedTracingModes.AI_AND_W3C)
+    .setInternalLogging(applicationInsights.internalLogging, true)
+    .start()
+
+  const roleName = process.env.npm_package_name
```

Comment:
do you think this is the most reliable way to get the package name?

Reply:
We only reference it in here don't we? I think its the standard way for pulling in params from the _package.json_.